src/main/java/org/lightcouch/internal/URIBuilder.java

```diff
@@ -115,10 +115,37 @@ public URIBuilder pathToEncode(String path) {
         return this;
     }
 
-    public URIBuilder query(String name, Object value) {
-        if (name != null && value != null)
-            this.qParams.addParam(name, value.toString());
+    /**
+     * Add the given {@code name} and {@code value} to the query parameters or update
+     * the existing query parameter matching {@code name} with the new {@code value}.
+     * @param name The name to add/update.
+     * @param value The value to add/update.
+     * @param update set to true to update an existing query parameter matching {@code name},
+     *               or false to add a new one. Note that if this is true and there is no
+     *               parameter matching {@code name}, the parameter will be added.
+     * @return The updated {@link URIBuilder} object.
+     */
+    public URIBuilder query(String name, Object value, boolean update) {
+        if (name != null && value != null) {
+            if (update) {
+                this.qParams.updateOrAdd(name, value.toString());
```

Comment:
I don't know enough about queries with Cloudant to know if it ever would make sense to have multiple query parameters with the same name?  If it does, then you wouldn't want to update, but would want to add in those cases. I kind of assumed that there could be cases when it's necessary to have multiple query parameters with the same name and different values partly because `Params` stores the parameters internally using a list (whereas I think it would be more sensible to use a map if the keys are unique).  If we're sure that this is the way to go, we should also change `Params` to use a map.